src/main/java/net/imglib2/img/basictypeaccess/buffer/LongBufferLongAccess.java

```diff
@@ -0,0 +1,104 @@
+package net.imglib2.img.basictypeaccess.buffer;
+
+import java.nio.ByteBuffer;
+import java.nio.LongBuffer;
+
+import net.imglib2.img.basictypeaccess.LongAccess;
+import net.imglib2.img.basictypeaccess.array.ArrayDataAccess;
+import net.imglib2.img.basictypeaccess.volatiles.VolatileAccess;
+
+public class LongBufferLongAccess implements ArrayDataAccess< LongBufferLongAccess >, LongAccess, VolatileAccess
+{
+
+	// TODO How to properly design a buffer access?
+	// 1. Implementing ArrayDataAccess probably makes sense
+	// 2. Should we have separate accesses that implement VolatileAccess?
+	// 3. Should we use ByteBuffer only or use specific buffers like LongBuffer.
+	//    allocateDirect is only available in ByteBuffer.
+	//    Both accesses could be implemented, i.e. LongBufferLongAccess as well
+	//    as ByteBufferLongAccess
+
+	private static boolean DEFAULT_IS_VALID = true;
+
+	private static int NUM_BYTES = Long.BYTES;
+
+	private final LongBuffer buffer;
+
+	private final boolean isValid;
+
+	public LongBufferLongAccess( final int numEntities )
+	{
+		this( numEntities, DEFAULT_IS_VALID );
+	}
+
+	public LongBufferLongAccess( final int numEntities, final boolean isValid )
+	{
+		this( LongBuffer.allocate( numEntities ), isValid );
+	}
+
+	public LongBufferLongAccess( final ByteBuffer buffer )
+	{
+		this( buffer, DEFAULT_IS_VALID );
+	}
+
+	public LongBufferLongAccess( final ByteBuffer buffer, final boolean isValid )
+	{
+		this( buffer.asLongBuffer(), isValid );
+	}
+
+	public LongBufferLongAccess( final LongBuffer buffer )
+	{
+		this( buffer, DEFAULT_IS_VALID );
+	}
+
+	public LongBufferLongAccess( final LongBuffer buffer, final boolean isValid )
+	{
+		this.buffer = buffer;
+		this.isValid = isValid;
+	}
+
+	@Override
+	public boolean isValid()
+	{
+		return isValid;
+	}
+
+	@Override
+	public long getValue( int index )
+	{
+		return buffer.get( index );
+	}
+
+	@Override
+	public void setValue( int index, long value )
+	{
+		buffer.put( index, value );
+
+	}
+
+	@Override
+	public LongBufferLongAccess createArray( int numEntities )
+	{
+		// TODO
+		// Should we respect the kind of buffer that this.buffer is?
+		// i.e. if this.buffer is direct, then the buffer of the new
+		// access should also be direct? Is there a method method in
+		// Buffer that allows to do that?
+		return buffer.isDirect()
+				? new LongBufferLongAccess( ByteBuffer.allocateDirect( numEntities * NUM_BYTES ), isValid )
+				: new LongBufferLongAccess( numEntities, isValid );
+	}
+
+	@Override
+	public LongBuffer getCurrentStorageArray()
+	{
+		return buffer;
+	}
+
+	@Override
+	public int getArrayLength()
+	{
+		return buffer.capacity();
```

Comment:
```suggestion
		return buffer.limit();
```

`getArrayLength()` should be `limit()` because that defines practical length of the data in `Buffer`.

https://docs.oracle.com/javase/8/docs/api/java/nio/Buffer.html

> A buffer's capacity is the number of elements it contains. The capacity of a buffer is never negative and never changes.
> 
> A buffer's limit is the index of the first element that should not be read or written. A buffer's limit is never negative and is never greater than its capacity.

`buffer.get(int index)` throws a `IndexOutOfBoundsException` if `index >= buffer.limit()`
https://docs.oracle.com/javase/8/docs/api/java/nio/ByteBuffer.html#get-int-

> IndexOutOfBoundsException - If index is negative or not smaller than the buffer's limit